### DISCUSSION TITLE
Read values from Smart Meter ITRON 3.HZ with ESP8266

### DISCUSSION THREAD

**Discussion by Tom0815-newbie** [arendst/Tasmota#22690] [Support request]
Hello everybody. 

I´m really a new on this topic.
But as already mentioned in the description, I´m trying to receive actual values from ITRON 3.HZ smart meter.
I´m using an ESP8266 nodemcu with tsop4838 ir receiver. 
Tasmota is already flashed on the device and it shows me the webui.
To flash the device with tasmota I used visual studio code.
I implemented the use SML script in the user_config_override.h as described on the page.

Unfortunately I dont see any values on the tasmota webui:
![WebUI](https://github.com/user-attachments/assets/7912ea3a-23db-4103-874e-e4187c8241aa)

If I use sensor53 d1, I can see data coming out of the smart meter. 
They all start with 77 and a lot of other numbers.
All starting with 77 XX XX XX ....

This is my script, which I copied from the smart meter tasmota page:
![Script](https://github.com/user-attachments/assets/ae658a58-6529-460e-af50-d0351e4da013)

I have already entered my PIN in the smart meter.
And the Inf is switched to ON.
Unfortunately both did not help.

What am I doing wrong?

**Comment by spueli99** [comment]
Hi, why don't you just use a ttl read head?
https://wiki.volkszaehler.org/hardware/controllers/ir-schreib-lesekopf-ttl-ausgang

You could also try to decode a dump.
https://tasmota-sml-parser.dicp.net/

**Comment by joba-1** [comment]
spueli99 is right, your tsop4838 is for receiving signals from IR remote controls encoded in 38kHz signals. 
What you need is more simple. An IR LED and an NPN transistor is enough.

I have the same Itron and use just that with my own firmware where I describe the connections:
https://github.com/joba-1/ElectricityMeter

A Volkszähler does the same thing and will also work.

**Comment by Tom0815-newbie** [comment]
Hello everyone,

First of all: thank you for your answers.

This forum is perfect for all tinkers.

@spueli99: I have just read through all the information you gave me. Unfortunately I don't have an IR read/write head and I am not able to build such a cool circuit board.
Apart of that, do I have to PULL the ITRON to get information? I thought the device PUSHES the information automatically as soon as I enter the PIN ?
![image](https://github.com/user-attachments/assets/a4bcd19a-ed51-4891-a99a-91acb4f45c11)
I additionally used the parser but unfortunately I dont get any results. Actually my values look like this (Sensor53 d1):
![image](https://github.com/user-attachments/assets/fe9081ac-3e6b-4b81-beff-4d68bc94d502)
It seems, that something is send, but I cant read it...

@joba-1: You are absolutely right. The TSOP4838 is an old one from a TV ;-). I thought I could use that too. I did not expect, that there are special IR LED to use. What does 38kHz mean? What frequency do I have to use? 
I mean, I get some information, which can be read by the TSOP4838. 

Can you please give me some advice, which IR LED  I need? Do I need also an IR Transceiver (PUSH)? Apart of that, do I need any other devices (resistor or capacitor)? I´m looking for a low cost variant ;-).
Maybe you have also a wiring diagram for the parts?

Sorry for all of these questions, but my school time is long time ago, however I feel like tinkering around a bit.

Many thanks in advance.

**Reply by joba-1** [reply]
> Unfortunately I don't have an IR read/write head

you can buy them. They are not expensive and they also solve the problem of fixing the led in front of the itron. 

Push is correct. If you put a camera in front of the led you should be able to see the itron led blink already.

> I did not expect, that there are special IR LED to use

It is the other way round. Your tsop is special and you need the more simple leds that don’t do any decoding themselves. 

> do I need any other devices

No other electronic parts are required.
It is the lowest cost solution but requires a bit more tinkering. 
There already is a kind of ascii circuit diagram at the link I gave but I can make a more common one if needed.

The two needed parts are really very generic, many will do, but to give you explicit  examples I‘ll add links:

* https://www.reichelt.de/de/de/shop/produkt/infrarot-diode_gaalas_940_nm_50_5_mm_t-1_3_4-219706
* https://www.reichelt.de/de/de/shop/produkt/bipolartransistor_npn_65v_0_1a_0_5w_to-92-238962

**Comment by Tom0815-newbie** [comment]
Hello together,

unfortunately I was ill quite a few days, so that this project needed to be postponed.
But now I´m back and I ordered/received the parts.
Can you please help me, how to wire the parts? 
I dont know how to connect the transistor to the IR Diode and where to connect the ESP8266 nodemcu.

Apart of that, it seems, that the diode is emitting also IR light. 
What am I supposed to do with that? 
I wanted to read the values coming from the ITRON Smart Meter.
So I expeceted to need an IR Fototransistor or so. 
Or did I understand something complete wrong?

Many thanks for you help.

**Reply by joba-1** [reply]
Ill is always bad, good you are back on track :)

All leds emit light, but that is not needed here. 

How to connect depends on the exact transistor you have. The one from my link?

**Reply by Tom0815-newbie** [reply]
Hallo joba-1,

yes, I bought the parts from your links. The NPN Transistor and the IR diode.
If an emitting IR diode is not usefull, I also have a Fototransistor here:
https://www.reichelt.de/de/de/shop/produkt/fototransistor_npn_730_1120nm_24_tht-3mm-60553
Is this one better?
But my question regarding the wiring still remains 😅.

**Reply by joba-1** [reply]
Could be that this transistor is even easier to use. It is effectively an ir led and a transistor in one housing.

If it works depends on the exact details from the datasheet. I‘ll have a look…

**Reply by joba-1** [reply]
wiring from my link 6 months ago…

GND -- -_LED_+ -- NPN_B

GND -- E_NPN_C -- R_10k -- V_3.3

NPN_C -- GPIO_Rx

e.g. the last line means: connect C_ollector of your NPN transistor to the GPIO pin of the ESP that you configure as serial Rx pin

**Reply by joba-1** [reply]
[how to identify - pin (cathode) of the led](https://de.m.wikipedia.org/wiki/Datei:Uvled_highres_macro.jpg)

**Reply by joba-1** [reply]
identify C (collector), E (emitter) and B (base) pin of your transistor. Image top left on the first page:

https://cdn-reichelt.de/documents/datenblatt/A100/BC546_48-CDIL.pdf

**Reply by joba-1** [reply]
use the photo transistor as one side of a voltage divider:

![image](https://github.com/user-attachments/assets/dbd7a62d-ffbb-4c63-8ce6-9ab1f30286fb)

image from a [beginner tutorial](https://www.build-electronic-circuits.com/phototransistor/)

* vcc is the 3.3V from your ESP
* vout is your Rx Gpio
* triangle lines at the bottom is ground from your ESP
* zig zag line is a 10k ohm resistor

if you don’t have a 10k resistor: anything from 3k to 30k should work just fine.

**Reply by joba-1** [reply]
I just noticed, if you use the photo transistor you need to switch the resistor and the transistor or the serial signal would be inverted…

**Comment by Tom0815-newbie** [comment]
Hello Joba-1,

first of all, many thanks for your fast replies.
I´m learning a lot during this project ;-).

Do I understand your wiring instructions correctly?

GND -- -LED+ -- NPN_B
GND -- E_NPN_C -- R_10k -- V_3.3
NPN_C -- GPIO_Rx

I tried to make a small sketch.
![image](https://github.com/user-attachments/assets/68881c9a-7191-4528-8283-35cc0231d565)
Can you please confirm this wiring?
